### PR TITLE
Fix/workout creation interface

### PR DESCRIPTION
**Pull request by Siupahc**
# Features

- For the workout creation, the UI doesn't look completely trash now. I created an ExerciseCard composable, which is in the composables package, so the list of exercies is actually not ugly and match the figma.
- I also created an abstract class called Exercise, because in our implementation, YogaExercise and BodyWeightExercise were like independant, but they are very much the same, and this makes things easier when trying to deal with generics and stuff. For the same purpose, I created an interface caled ExerciseType, to "link" YogaExerciseType and BodyWeightExerciseType. Both can be found in **Exercise.kt** in the model.workout package. This is what I mentioned in #35 
- Used a LazyColumn in the WorkoutCreationScreen, in order to be able to scroll if you have a lot of exercises. Without this, you would effectively be stuck, this is now fixed.

# Others

Made my code a little bit less messy, and added a simple test for my new composable.

### CODE REVIEW

File: app/src/main/java/com/android/sample/ui/workout/WorkoutCreationScreen.kt [hunk 1]
```diff
@@ -114,69 +136,113 @@ fun WorkoutCreationScreen(
...
-                        navigationActions.navigateTo(Screen.MAIN)
-                      },
-                      modifier = Modifier.padding(16.dp).testTag("saveButton")) {
-                        Text("Save")
-                      }
+                    Card(
+                        shape = RoundedCornerShape(16.dp),
+                        colors =
+                            CardDefaults.cardColors(
+                                containerColor = Color(0xFFD3D3D3)), // Couleur grise pour la carte
```
Comment (AlecLeGrooker) [inline app/src/main/java/com/android/sample/ui/workout/WorkoutCreationScreen.kt:195]: Same for Color

File: app/src/main/java/com/android/sample/ui/workout/WorkoutCreationScreen.kt [hunk 2]
```diff
@@ -114,69 +136,113 @@ fun WorkoutCreationScreen(
...
+                                .padding(horizontal = 24.dp, vertical = 8.dp)) {
+                          // Box cliquable avec texte "+"
+                          Box(
+                              contentAlignment = Alignment.Center,
+                              modifier =
+                                  Modifier.fillMaxWidth()
+                                      .height(56.dp)
+                                      .clickable { showExerciseDialog = true }
+                                      .testTag("addExerciseButton")) {
+                                Text(text = "+", fontSize = 24.sp, color = Color.DarkGray)
```
Comment (AlecLeGrooker) [inline app/src/main/java/com/android/sample/ui/workout/WorkoutCreationScreen.kt:207]: We should avoide strings defined like that. The preferable way is to define it in string.xml like that : 
<string name="addButton">+</string> and replace text = "+" by text = stringResource(id = R.string.addButton)

**Review by AlecLeGrooker — COMMENTED** [review]
Great job overall ! You really improved the look and the functionalitly of the creation. Just a few fix should not be long.

File: app/src/main/java/com/android/sample/ui/theme/Color.kt [hunk 3]
```diff
@@ -6,7 +6,9 @@ val Purple80 = Color(0xFFD0BCFF)
 val PurpleGrey80 = Color(0xFFCCC2DC)
 val Pink80 = Color(0xFFEFB8C8)
 
+val Purple60 = Color(0xFF9C7EEA)
 val Purple40 = Color(0xFF6650a4)
+val Purple20 = Color(0xFFEAE6FF)
```
Comment (didi2101) [inline app/src/main/java/com/android/sample/ui/theme/Color.kt:11]: Colors have been fixed

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -5,4 +5,5 @@
     <string name="title_activity_second">SecondActivity</string>
     <string name="welcome_message">What\'s up %1$s?</string>
     <string name="setting_title">Settings</string>
+    <string name="addButton">+</string>
```
Comment (didi2101) [inline app/src/main/res/values/strings.xml:8]: And the name too

**Review by didi2101 — APPROVED** [review]
Comments have been taken into account.

Approved.

**Pull request merged**